common-core-open/src/main/java/edu/isi/nlp/files/KeyValueSources.java

```diff
@@ -92,4 +93,26 @@ private KeyValueSources() {
     }
     return new ZipKeyValueSource(zipFile, ret.build());
   }
+
+  @Nonnull
+  public static ImmutableKeyValueSource<Symbol, ByteSource> fromVistaUtilsZipKeyValueSource(
+      final ZipFile zipFile) {
+    // read "__keys" file
+    ZipEntry keysEntry = zipFile.getEntry(("__keys"));
+    String keysString = null;
+    try {
+      keysString = ZipFiles.entryAsString(zipFile, keysEntry, StandardCharsets.US_ASCII);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    String[] keysArray = keysString.split("\n");
+
+    // explicitly construct the second argument to the ZipKeyValueSource constructor
+    final ImmutableMap.Builder<Symbol, String> keyMapBuilder = ImmutableMap.builder();
+    for (int i = 0; i < keysArray.length; i++) {
+      String key = keysArray[i];
+      keyMapBuilder.put(Symbol.from(key), key);
+    }
+    return new ZipKeyValueSource(zipFile, keyMapBuilder.build());
+  }
```

Comment:
Also, please add a test.  (You can make a tiny example zip on the Python side to commit to this repo)